tests/hexadecimal-utils/test_is_hexstr.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from eth_utils import is_hexstr
+
+
+@pytest.mark.parametrize(
+    "value,expected",
+    (
+        ("", False),  # empty string
+        ("0x", True),  # empty string
+        ("0X", True),  # empty string
+        ("abcdef1234567890", True),
+        ("ABCDEF1234567890", True),
+        ("AbCdEf1234567890", True),
+        ("0xabcdef1234567890", True),
+        ("0xABCDEF1234567890", True),
+        ("0xAbCdEf1234567890", True),
+        ("12345", True),  # odd length
+        ("0x12345", True),  # odd length
+        ("123456xx", False),  # non-hex character
+        ("0x123456xx", False),  # non-hex character
+        ("0\u0080", False),  # triggers different exceptions in py2 and py3
+        (1, False),  # int
+        (b"", False),  # bytes
+        ({}, False),  # dicitionary
```

Comment:
Tiny typo: `dictionary`, but these tests look overall pretty solid. 

It seems like line 11 should cover it, but it might be worth adding a line 17 duplicate that's `"0X"` prefixed.